src/classes/DML.cls

```diff
@@ -0,0 +1,53 @@
+/*************************************************************************************************
+* This file is part of the Nebula Framework project, released under the MIT License.             *
+* See LICENSE file or go to https://github.com/jongpie/NebulaFramework for full license details. *
+*************************************************************************************************/
+public virtual class DML implements IDML {
+    public virtual SObject doInsert(SObject record) { return this.doInsert(new List<SObject>{record})[0];}
```

Comment:
@jongpie  this was a change suggested by Marty to come in line with the return types for the actual DML statements themselves.  I agree, typically you wouldn't actually be re-updating your variables to be equal to what is being returned.  More of being consistent with the larger salesforce API